<h1 style="padding:8px; margin:0px -20px; color:#FFF; background:#06D">
Sang Data Analysis
</h1>

- AFTISSE Melissa (<aftisse.melissa@etu.u-bordeaux.fr>)
- RAVENDIRANE Gayathiri (<gayathiri.ravendirane@etu.u-bordeaux.fr>)

Version : 1.0

In [2]:
# IMPORTS
import numpy as np
import seaborn as sns
import pandas as pd 
import matplotlib.pyplot as plt
from ipywidgets import interact, interact_manual 


<h2 style="padding:10px; color:#FFF; background:#06D">Importation du jeu de données</h2>

In [14]:
HypoRiskModel_kaggle = pd.read_csv("https://gayathiri-ravendirane.emi.u-bordeaux.fr/DATA/HypoRiskModel_kaggle.csv")
print(HypoRiskModel_kaggle.shape)
HypoRiskModel_kaggle.head()

(4240, 13)


male  age  currentSmoker  cigsPerDay  BPMeds  diabetes  totChol  sysBP  \
0     1   39              0         0.0     0.0         0    195.0  106.0   
1     0   46              0         0.0     0.0         0    250.0  121.0   
2     1   48              1        20.0     0.0         0    245.0  127.5   
3     0   61              1        30.0     0.0         0    225.0  150.0   
4     0   46              1        23.0     0.0         0    285.0  130.0   

   diaBP    BMI  heartRate  glucose  Risk  
0   70.0  26.97       80.0     77.0     0  
1   81.0  28.73       95.0     76.0     0  
2   80.0  25.34       75.0     70.0     0  
3   95.0  28.58       65.0    103.0     1  
4   84.0  23.10       85.0     85.0     0

In [17]:
BloodPressure_kaggle = pd.read_csv("https://gayathiri-ravendirane.emi.u-bordeaux.fr/DATA/BloodPressure_kaggle.csv")
print(BloodPressure_kaggle.shape)
BloodPressure_kaggle.head()

(2000, 15)


Patient_Number  Blood_Pressure_Abnormality  Level_of_Hemoglobin  \
0               1                           1                11.28   
1               2                           0                 9.75   
2               3                           1                10.79   
3               4                           0                11.00   
4               5                           1                14.17   

   Genetic_Pedigree_Coefficient  Age  BMI  Sex  Pregnancy  Smoking  \
0                          0.90   34   23    1        1.0        0   
1                          0.23   54   33    1        NaN        0   
2                          0.91   70   49    0        NaN        0   
3                          0.43   71   50    0        NaN        0   
4                          0.83   52   19    0        NaN        0   

   Physical_activity  salt_content_in_the_diet  alcohol_consumption_per_day  \
0              45961                     48071                          NaN   
1              26106                     25333                        205.0   
2               9995                     29465                         67.0   
3              10635                      7439                        242.0   
4              15619                     49644                        397.0   

   Level_of_Stress  Chronic_kidney_disease  Adrenal_and_thyroid_disorders  
0                2                       1                              1  
1                3                       0                              0  
2                2                       1                              0  
3                1                       1                              0  
4                2                       0                              0

In [20]:
BloodValues_kaggle = pd.read_excel("https://gayathiri-ravendirane.emi.u-bordeaux.fr/DATA/BloodValues_kaggle.xlsx")
print(BloodValues_kaggle.shape)
BloodValues_kaggle.head()

(4598, 14)


Gender (0:Female, 1: Male)  Age  Glucose (fasting)  \
0                           0   59                 90   
1                           0   69                144   
2                           0   63                314   
3                           0   63                228   
4                           1   66                 97   

   AST (Aspartat Aminotransferaz)  ALT (Alanin Aminotransferaz)  Creatinine  \
0                              14                            12        0.81   
1                              19                            14        0.75   
2                              11                            17        0.86   
3                               7                            17        0.81   
4                              18                            21        0.98   

   blood urea nitrogen (BUN)  blood urea nitrogen - [ÜRE]  \
0                         11                        23.54   
1                         16                        34.24   
2                         12                        25.68   
3                         21                        44.94   
4                         24                        51.36   

   Thyroid Stimulating Hormone (TSH)  Triglyceride  Cholesterol  \
0                              0.979           109          196   
1                              2.420           107          178   
2                              7.690           443          188   
3                              3.860           161          248   
4                              0.414            84          164   

   High Density Lipoprotein (HDL Cholesterol)  \
0                                        67.3   
1                                        46.1   
2                                        51.4   
3                                        42.2   
4                                        41.1   

   Low Density Lipoprotein (LDL Cholesterol)  Free T4 (FT4)  
0                                        107          1.590  
1                                        111          0.935  
2                                         77          1.320  
3                                        174          1.280  
4                                        106          1.290

In [22]:
BloodGroupWWW_kaggle = pd.read_csv("https://gayathiri-ravendirane.emi.u-bordeaux.fr/DATA/BloodGroupWWW_kaggle/cleaned_blood_type_distribution_by_country.csv")
print(BloodGroupWWW_kaggle.shape)
BloodGroupWWW_kaggle.head()

(126, 10)


Country/Dependency  Population      O+      A+      B+    AB+     O-     A-  \
0            Albania   3,074,579  34.10%  31.20%  14.50%  5.20%  6.00%  5.50%   
1            Algeria  43,576,691  40.00%  30.00%  15.00%  4.25%  6.60%  2.30%   
2          Argentina  45,479,118  50.34%  31.09%   8.20%  2.16%  4.29%  2.98%   
3            Armenia   3,021,324  29.00%  46.30%  12.00%  5.60%  2.00%  3.70%   
4          Australia  25,466,459  38.00%  32.00%  12.00%  4.00%  7.00%  6.00%   

      B-    AB-  
0  2.60%  0.90%  
1  1.10%  0.75%  
2  0.74%  0.20%  
3  1.00%  0.40%  
4  2.00%  1.00%

BILAN

- HypoRiskModel_kaggle
  - (4240, 13)
  - Points clés: variables liées à l’hypotension et facteurs de risque (âge, mesures cliniques). Utile pour créer des indicateurs de risque, segmenter des patients et entraîner des modèles prédictifs.

- BloodPressure_kaggle
  - (2000, 15)
  - Points clés: mesures de pression artérielle (systolique/diastolique) et métadonnées (p. ex. âge, sexe, habitudes). Utile pour la distribution des BP, la définition de seuils (normale/élevée/hypertension) et l’étude de corrélations.

- BloodValues_kaggle
  - (4598, 14)
  - Points clés: paramètres biologiques (valeurs sanguines) permettant des analyses de profils, la détection d’anomalies et le croisement avec BP/hypotension pour enrichir les features des modèles.

- BloodGroupWWW_kaggle (distribution des groupes sanguins par pays)
  - (126, 10)
  - Points clés: répartition des groupes sanguins (A, B, AB, O, Rh) par pays. Utile pour des cartes/charts comparatifs, analyses géographiques et mise en relation disponibilité/besoins médicaux.

<div style="padding:8px; margin:0px -20px; color:#FFF; background:#06D; text-align:right">● ● ● </div>